kernel/Kconfig

```diff
@@ -760,6 +760,21 @@ config TICKLESS_KERNEL
 	  This option enables a fully event driven kernel. Periodic system
 	  clock interrupt generation would be stopped at all times.
 
+config USLEEP
```

Comment:
I think you can skip this kconfig.  There's no cost if this is enabled but unused, and it doesn't change the internals of anything.  It's just an extra API.  The linker will drop it for us.

Reply:
TIL that linkers are way smarter than they used to be.  Hmph.

Reply:
There's another reason for the config option though.  See my comment to @nashif below.  The "other work planned" is, I believe, something you've got at least a hand in. :)